personaelabs/nym
# API updates

## Summary

Update the API to address issues mentioned in [Tony’s notes](https://www.notion.so/0cf645411c3a4f44a264daab2824ec23)

### Notable changes
- Return the entire upvote model instead of returning the upvote count.
- Use Prisma’s type inference for each GET route.
- Unify the `NymPost` and `DoxedPost` models into a single `Post` model.
    - This was required to do recursive queries of replies using Prisma’s native methods (i.e. without writing a raw query)
- Add constraints to columns
    - Check that “id”s are prefixed with “0x” (following nymjs’s specification)
    - Add a 1-to-n foreign key relation from `Post` to `Tree`, and `DoxedUpvote` to `Tree`.

## How to test
1. Populate test data by running `pnpm test_data:populate` in the root of the repo.
2. Run `pnpm frontend:dev` in the root of the repo and 
   1. navigate to "http://localhost:3000/api-doc". Call the GET methods and check the responses.
   2. navigate to "http://localhost:3000/example". Connect with the account I shared in Discord, and test "Submit a doxed post" and "Upvote a post". Necessary info is console.loged. ("Submit a pseudonymous post" is currently broken, but it'll likely take a bit of time to fix, and I believe it's not strictly necessary to proceed in building UI components, so I'll postpond the fix to a future PR.)

## Future PRs
- Fix Nym post creation in pages/example.tsx
    - Due to some wasm issues in spartan-ecdsa,  the Nym post submission in the example page is currently broken.
